### DISCUSSION TITLE
Collection was modified; enumeration operation may not execute

### DISCUSSION THREAD

**Discussion by mgth** [AvaloniaUI/Avalonia#13754] [Q&A]
## Describe the bug
Collection was modified; enumeration operation may not execute in Avalonia.Base.dll at BuildAvaloniaApp()
            .StartWithClassicDesktopLifetime(args) level.

Random crash using ListBox connected to an observable collection

seams to append on highlight changing (but i might be wrong), even if observed collection did not changed.

## To Reproduce
Hard to tell, but its quite rare.

## Environment 

- OS: Windows 11
- Avalonia-Version: 11.0.5

**Comment by timunie** [comment]
Do you do some async operations when pooulatiing the data? Best would be to have some code or sample to reproduce.

**Comment by maxkatz6** [comment]
Make sure you update UI-bound collections on the UI thread.

**Comment by timunie** [comment]
I'm converting this into a discussion for now. As soon as we have a repo and find it is a bug, I'll open the issue again.
